File: python/tests/bits_needed_tests.py
import sys
sys.path.append('..')

from util import bits_needed


def test_bits_needed():
    def check(expected, n):
        actual = bits_needed(n)
        assert expected == actual
    yield check, 1, 2
    yield check, 2, 3
    yield check, 2, 4
    yield check, 3, 5
    yield check, 8, 2**8

File: cpp/Huffman/Huffman/tests/bits-needed-tests.cpp
#ifdef TEST_BUILD

#include "util.h"
#include "catch.hpp"


namespace
{
    void check(u64 domain_size, unsigned expected)
    {
        int actual = bits_needed(domain_size);

        CHECK(actual == expected);
    }
}

#define TEST(DS, E) TEST_CASE("bits_needed(" #DS ") -> " #E) { check(DS, E); }

TEST(2, 1)
TEST(3, 2)
TEST(4, 2)
TEST(5, 3)
TEST(6, 3)
TEST(7, 3)
TEST(8, 3)
TEST(9, 4)
TEST(128, 7)
TEST(129, 8)
TEST(1'000'000, 20)

#endif
